sockjs/websocket.go

```diff
@@ -1,23 +1,32 @@
 package sockjs
 
 import (
-	"fmt"
 	"net/http"
-	"strings"
 
 	"github.com/gorilla/websocket"
 )
 
 // WebSocketReadBufSize is a parameter that is used for WebSocket Upgrader.
 // https://github.com/gorilla/websocket/blob/master/server.go#L230
+//
+// Deprecated: Set WebSocketUpgrader.ReadBufferSize instead.
 var WebSocketReadBufSize = 4096
 
 // WebSocketWriteBufSize is a parameter that is used for WebSocket Upgrader
 // https://github.com/gorilla/websocket/blob/master/server.go#L230
+//
+// Deprecated: Set WebSocketUpgrader.WriteBufferSize instead.
 var WebSocketWriteBufSize = 4096
 
+// WebSocketUpgrader is used to configure websocket handshakes and websocket
+// connection details.
+var WebSocketUpgrader = &websocket.Upgrader{
+	ReadBufferSize:  WebSocketReadBufSize,
+	WriteBufferSize: WebSocketWriteBufSize,
+}
+
 func (h *handler) sockjsWebsocket(rw http.ResponseWriter, req *http.Request) {
-	conn, err := websocket.Upgrade(rw, req, nil, WebSocketReadBufSize, WebSocketWriteBufSize)
+	conn, err := WebSocketUpgrader.Upgrade(rw, req, nil)
```

Comment:
I think here can be some backwards compatibility issues, in Gorilla Websocket exported function `Upgrade` also sets `Error` and `CheckOrigin` [Upgrader fields](https://github.com/gorilla/websocket/blob/master/server.go#L262), so looks like `WebsocketUpgrader` should look like:

```go
var WebSocketUpgrader = &websocket.Upgrader{
	ReadBufferSize:  WebSocketReadBufSize,
	WriteBufferSize: WebSocketWriteBufSize,
        Error: func(w http.ResponseWriter, r *http.Request, status int, reason error) {
		// don't return errors to maintain backwards compatibility
	},
        CheckOrigin: func(r *http.Request) bool {
		// allow all connections by default
		return true
	}
}
```

Also previously it could be possible at any moment override `sockjs.WebSocketReadBufSize` and `sockjs.WebSocketWriteBufSize` so new values will be used for new connections upgrade  - now it's also possible but only via:

```go
sockjs.WebSocketUpgrader.ReadBufferSize = 0
```

Modifying `WebSocketReadBufSize` and `WebSocketWriteBufSize` won't take an effect for new conns. 

Actually as far as I understand it's now better to have `0` as read and write buffer default values so Gorilla Websocket could [reuse buffers returned from Hijack method](https://github.com/gorilla/websocket/blob/master/server.go#L30). Maybe setting `WebSocketReadBufSize` and `WebSocketWriteBufSize` to `0` will be a desirable feature as it ams to reduce memory allocs. But it's still a bit backwards incompatible.